daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/Dar.scala

```diff
@@ -3,6 +3,20 @@
 
 package com.digitalasset.daml.lf
 
+import scalaz.{Functor, Semigroup}
+
 case class Dar[A](main: A, dependencies: List[A]) {
-  def all: List[A] = main :: dependencies
+  lazy val all: List[A] = main :: dependencies
+}
+
+object Dar {
+  implicit def darSemigroup[A]: Semigroup[Dar[A]] = new Semigroup[Dar[A]] {
```

Comment:
this instance is not very good, i think. the main on the right magically becomes a non-main. the functor one is good.

Reply:
yep, that by design and it is documented in the test. This should be enough for the `scala/codegen` and I believe `java/codegen` both as of now don't need main.

Thinking about it a bit more... I can change the `Dar` so main semantic does not get lost after the append. Dar will become a variant, something like:
```
sealed trait Dar
case class Empty extends Dar
case class DarNode[A](main: A, dependencies: List[A]) extends Dar
case class DarList(list: NonEmptyList[DarNode])
```
with `Empty` it can also become a `Monoid`, right?

Reply:
but you know, I don't actually need to append the `Dar`, I have a map, so I can easily iterate over a list of `Dar`'s and `map` them with `scalaCodegen` one thing at a time.

Reply:
so, `Semigroup` is going away.

Reply:
great, can you change the pr to remove it?